src/components/AdminSettings/GeneralSettings.vue

```diff
@@ -27,15 +27,36 @@
 
 		<NcCheckboxRadioSwitch :model-value="isConversationsFilesChecked"
 			:disabled="loading || loadingConversationsFiles"
+			type="switch"
 			@update:model-value="saveConversationsFiles">
 			{{ t('spreed', 'Allow conversations on files') }}
 		</NcCheckboxRadioSwitch>
 
 		<NcCheckboxRadioSwitch :model-value="isConversationsFilesPublicSharesChecked"
 			:disabled="loading || loadingConversationsFiles || !isConversationsFilesChecked"
+			type="switch"
 			@update:model-value="saveConversationsFilesPublicShares">
 			{{ t('spreed', 'Allow conversations on public shares for files') }}
 		</NcCheckboxRadioSwitch>
+
+		<h3>
+			{{ t('spreed', 'End-to-end encrypted calls') }}
+			<small>{{ t('spreed', 'Beta') }}</small>
+		</h3>
+
+		<NcCheckboxRadioSwitch v-model="isE2EECallsEnabled"
+			type="switch"
+			:disabled="loading || !canEnableE2EE"
+			@update:model-value="updateE2EECallsEnabled">
+			{{ t('spreed', 'Enable encryption') }}
+		</NcCheckboxRadioSwitch>
+
+		<NcNoteCard v-if="!canEnableE2EE"
+			type="warning"
+			:text="t('spreed', 'End-to-end encryption is only possible with a High-performance backend.')" />
```

Comment:
This is a bit misleading. Without the HPB, the clients will create PeerConnections between themselves, which by definition then are also end-to-end encrypted.

Reply:
> Without the HPB, the clients will create PeerConnections between themselves, which by definition then are also end-to-end encrypted.

Well they shouldn't as we don't know if the other client can do it?
Maybe we need another call later today

Reply:
Summary from the call:
- Without HPB we shouldn't show the setting as it's "default"
- Without SIP we don't need to check if the HPB was updated
- If the setting is enabled, older mobile clients need to be blocked